internal/provider/resource_vault_secrets_secret.go

```diff
@@ -0,0 +1,176 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func NewVaultSecretsSecretResource() resource.Resource {
+	return &resourceVaultsecretsSecret{}
+}
+
+type resourceVaultsecretsSecret struct {
+	client *clients.Client
+}
+
+type Secret struct {
+	ID          types.String `tfsdk:"id"`
+	AppName     types.String `tfsdk:"app_name"`
+	SecretName  types.String `tfsdk:"secret_name"`
+	SecretValue types.String `tfsdk:"secret_value"`
+}
+
+func (r *resourceVaultsecretsSecret) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vault_secrets_secret"
+}
+
+func (r *resourceVaultsecretsSecret) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "The Vault Secrets secret resource manages a secret within a given application.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The id of the resource",
+				Computed:    true,
+			},
+			"app_name": schema.StringAttribute{
+				Description: "The name of the application the secret can be found in",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(3, 36),
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^[-\da-zA-Z]{3,36}$`),
+						"must contain only letters, numbers or hyphens",
+					),
+				},
+			},
+			"secret_name": schema.StringAttribute{
+				Description: "The name of the secret",
```

Comment:
It would probably be good to add validators here as well.

Reply:
Yeah. Good call. I'm not sure what is allowed for secret names. I'll have to double check.